java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java

```diff
@@ -107,6 +112,23 @@ public static void printResults(
     public static void testClientSetGet(
             Supplier<Client> clientCreator, BenchmarkingApp.RunConfiguration config, boolean async) {
         for (int concurrentNum : config.concurrentTasks) {
+            // same as Executors.newCachedThreadPool() with a RejectedExecutionHandler for robustness
+            ExecutorService executor =
+                    new ThreadPoolExecutor(
+                            0,
+                            Integer.MAX_VALUE,
+                            60L,
+                            TimeUnit.SECONDS,
+                            new SynchronousQueue<Runnable>(),
+                            (r, poolExecutor) -> {
+                                if (!poolExecutor.isShutdown()) {
```

Comment:
if the pool is called during shutdown, isn't that bug? should we silently ignore it?

Reply:
good point.  It needs to be re-thrown.  Thanks and fixed.